cmd/geth/main.go

```diff
@@ -325,6 +328,24 @@ func prepare(ctx *cli.Context) {
 			log.Info("Bumping default cache on mainnet", "provided", ctx.Int(utils.CacheFlag.Name), "updated", 4096)
 			ctx.Set(utils.CacheFlag.Name, strconv.Itoa(4096))
 		}
+	} else if ctx.String(utils.SyncModeFlag.Name) != "light" && !ctx.IsSet(utils.CacheFlag.Name) {
+		// We haven't set the cache, but may have set the network ID or the OP network flag we may be on an OP stack mainnet.
+		// This code is called prior to creating the new node, so we need to manually inspect several different flags.
+		// First pull from the network flag (which overrides the network ID flag), then look at the network ID flag if
+		// the named network is not used.
+		if ctx.IsSet(utils.BetaOPNetworkFlag.Name) && strings.Contains(ctx.String(utils.BetaOPNetworkFlag.Name), "mainnet") {
+			log.Info("Bumping default cache on mainnet", "provided", ctx.Int(utils.CacheFlag.Name), "updated", 4096, "network", ctx.String(utils.BetaOPNetworkFlag.Name))
+			ctx.Set(utils.CacheFlag.Name, strconv.Itoa(4096))
+		} else if ctx.IsSet(utils.NetworkIdFlag.Name) {
+			// Lookup a chain config & see if the superchain (L1 base chain) is Ethereum mainnet
+			// If we can't find the config or it's not mainnet, don't update the cache
+			chainID := ctx.Uint64(utils.NetworkIdFlag.Name)
+			chConfig, ok := superchain.OPChains[chainID]
+			if ok && chConfig.Superchain == "mainnet" {
+				log.Info("Bumping default cache on mainnet", "provided", ctx.Int(utils.CacheFlag.Name), "updated", 4096)
+				ctx.Set(utils.CacheFlag.Name, strconv.Itoa(4096))
+			}
+		}
```

Comment:
I don't think we should conflate network ID and chain ID - they're two separate values. While they are usually the same value in Ethereum, they have quite distinct roles and particularly since op-geth doesn't use devp2p much the network id is a particularly unreliable signal.

Reply:
Yea, we could remove this check. In some of our infra, we do set network ID , but maybe don't need to.